cadl-extension/src/main/java/com/azure/autorest/CadlPlugin.java

```diff
@@ -130,10 +123,6 @@ public String handlePartialUpdate(String filePath, String generatedContent) {
         SETTINGS_MAP.put("required-parameter-client-methods", true);
         SETTINGS_MAP.put("generic-response-type", true);
         SETTINGS_MAP.put("output-model-immutable", true);
```

Comment:
If we want to change, these are the things we don't want. In Cadl, we would like to make the `JavaSettings` instance without this map.
However the singleton nature of the `JavaSettings` make it hard.

Reply:
I see. I would not change that for now.